### PR TITLE
support for public key retrieval callback

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Updates the `verify` function to accept an optional `findKey` callback which can be used to retrieve the public key for verifying signatures which are not using a signing certificate (cases where verification material in the bundle specifies a public key hint instead of a certificate chain).

In a subsequent PR, the same key returned by `findKey` will also be passed-down to the tlog verification logic and used to re-create the Rekor entry for SET verification.

#### Release Note
* Updates the signature for the `verify` function to accept an optional `findKey` callback which will be invoked to retrieve the public key in scenarios where the signature was generated w/ a static keypair.

### CODE REVIEW

File: src/sigstore.ts [hunk 1]
```diff
@@ -43,7 +43,9 @@ export type SignOptions = {
   oidcClientSecret?: string;
 } & TLogOptions;
 
-export type VerifierOptions = TLogOptions;
+export type VerifierOptions = {
+  findKey?: FindKeyFunc;
```
Comment (feelepxyz) [inline src/sigstore.ts:47]: this is a very small nitpick, `find` sounds somehow out of place.. not entirely sure why, but wondering if `getKey` sounds better, or `getPublicKey` even? Like from the libs perspective, it's getting the key not finding it?

**Review by feelepxyz — DISMISSED** [review]
Thanks for adding this! 👍  had a very small nitpick on naming

File: src/sigstore.ts (hunk 1 above)
Comment (bdehamer) [reply src/sigstore.ts:47]: Good call. Updated!

File: src/sigstore.ts [hunk 2]
```diff
@@ -43,7 +43,9 @@ export type SignOptions = {
   oidcClientSecret?: string;
 } & TLogOptions;
 
-export type VerifierOptions = TLogOptions;
+export type VerifierOptions = {
+  getPublicKey?: FindKeyFunc;
```
Comment (feelepxyz) [inline src/sigstore.ts:47]: 😬 

```suggestion
  getPublicKey?: GetPublicKeyFunc;
```

Comment (bdehamer) [reply src/sigstore.ts]: 🤦🏻‍♂️

**Review by feelepxyz — APPROVED** [review]

**Pull request merged**